### ISSUE TITLE
Failing unit test in Node 10+ (connection timeout)

### ISSUE THREAD

**Issue by jancurn** [TooTallNate/node-https-proxy-agent#75]
The unit test of the `https-proxy-agent` package fails on Node 10+:

```
  1) HttpsProxyAgent "http" module should work over an HTTP proxy:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
  
  2) HttpsProxyAgent "http" module should work over an HTTPS proxy:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
  
  3) HttpsProxyAgent "after all" hook:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```

I suspect this is related to the following change introduced in Node 10: https://github.com/nodejs/node/pull/19981

Unfortunately, this issue makes it impossible to use web sockets over HTTPS proxy, since that depends on `https-proxy-agent`. For example, unit tests of https://github.com/apifytech/proxy-chain fail for this reason (see https://github.com/apifytech/proxy-chain/blob/master/test/server.js#L612)

**Comment by jancurn** [comment]
For a reference: As a workaround, rather than using `ws` + `https-proxy-agent` package to connect web socket via HTTPS proxy, one can use the `faye-websocket`package.

**Comment by TooTallNate** [comment]
Fixed by #73.
